src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -39,9 +39,50 @@ class Connection
      */
     private $amqpQueue;
 
-    public function __construct(array $connectionCredentials, array $exchangeConfiguration, array $queueConfiguration, bool $debug = false, AmqpFactory $amqpFactory = null)
+    /**
+     * @var \AMQPExchange|null
+     */
+    private $amqpRetryExchange;
+
+    /**
+     * Available options:
+     *
+     *   * host: Hostname of the AMQP service
+     *   * port: Port of the AMQP service
+     *   * vhost: Virtual Host to use with the AMQP service
+     *   * user: Username to use to connect the the AMQP service
+     *   * password: Username to use the connect to the AMQP service
+     *   * queue:
+     *     * name: Name of the queue
+     *     * routing_key: The routing key (if any) to use to push the messages to
+     *     * flags: Queue flags (Default: AMQP_DURABLE)
+     *     * arguments: Extra arguments
+     *   * exchange:
+     *     * name: Name of the exchange
+     *     * type: Type of exchange (Default: fanout)
+     *     * flags: Exchange flags (Default: AMQP_DURABLE)
+     *     * arguments: Extra arguments
+     *   * retry:
+     *     * attempts: Number of times it will try to retry
+     *     * routing_key_pattern: The pattern of the routing key (Default: "attempt_%attempt%")
+     *     * dead_queue: Name of the queue in which messages that retry more than attempts time are pushed to
+     *     * dead_routing_key: Routing key name for the dead queue (Default: "dead")
+     *     * queue_name_pattern: Pattern to use to create the queues (Default: "retry_queue_%attempt%")
```

Comment:
The only point of using different queues with each attempt is to be able to use different ttls, isn't it? But if the ttl is the same (the default), there is no point in having diferent queues. So to me, the retry queues should be based on the ttl, not the attempt number.

Reply:
That's correct, the different queues are used for different TTLs, but not only. On the operational side of the things (for whoever will monitor the RabbitMq), it's much clearer to understand what is going on in terms of retry, while it would be trickier with a `retry_in_X` IMHO)